app/templates/article.hbs

```diff
@@ -1,97 +1,92 @@
-<div class="post-page">
+<div class="article-page">
   <div class="banner">
     <div class="container">
-      <h1>How to build webapps that scale</h1>
-      <div class="post-meta">
-        <a href=""><img src="http://i.imgur.com/Qr71crq.jpg" alt="author image"></a>
-        <div class="info">
-          <a href="" class="author">Eric Simons</a>
-          <span class="date">January 20th</span>
-        </div>
-        <button class="btn btn-sm btn-outline-secondary">
-          <i class="ion-plus-round"></i> &nbsp; Follow Eric Simons
-          <span class="counter">(10)</span>
-        </button>
-        &nbsp;&nbsp;
-        <button class="btn btn-sm btn-outline-primary">
-          <i class="ion-heart"></i> &nbsp; Favorite Post
-          <span class="counter">(29)</span>
-        </button>
-      </div>
+      <h1>{{model.title}}</h1>
+      {{article-meta
+        article=model 
+        onFavoriteArticle=(action 
+          (if session.isLoggedIn 
```

Comment:
The `favorite` and `follow` buttons behave differently when the user is anonymous or logged in. I opted to do the logic check in the template as opposed to in the route actions themselves to keep the actions focused.

Reply:
I wrote above but is there a way to not use the `transition-to` helper and instead to use a `link-to`? Maybe inside the `article-meta` component?

Reply:
The only way I can think of is to add the template logic to use a `link-to` helper if the user is logged out. So the `article-meta` component template would be like

```handlebars
{{#if isLoggedIn}}
    <button {{action "onFavoriteArticle"}}>Favorite Article</button>
{{else}}
    {{#link-to "login"}}Favorite Article{{/link-to}}
{{/if}}
```

I'm okay with doing this too.

Reply:
I've added this.